src/Core/Employee/ContextEmployeeProviderInterface.php

```diff
@@ -51,4 +51,11 @@ public function getId();
      * @return int
      */
     public function getLanguageId();
+
+    /**
+     * Get context employee's profile ID.
+     *
+     * @return int
```

Comment:
Why not https://github.com/PrestaShop/PrestaShop/blob/81d54080c6e036c7f16120ae03c459ed39bf2820/src/Core/Domain/Profile/ValueObject/ProfileId.php ? Why mixin ProfileId int and class.. It's weird for user :/
ping @matks

Reply:
I'm not sure to be honest. Maybe the reason is that it does not come from the `Domain`. It's more like application layer service. But again, I may be wrong here.

Reply:
@PierreRambaud Indeed we have introduced ValueObjects to be used as inputs and outputs for CQRS. So far their usage has been restricted to this usecase and Adapter providers have been returning php primitives as does the legacy code.

We could start using ValueObjects everywhere indeed 🤔but then I would like to do it in 1.7.7 rather than in 1.7.6, in order to stay consistent.

Btw @PierreRambaud do you like our usage of ValueObjects ? 😁

Reply:
Not really, keep thinking it's really overkill to have a `getSomethindId()->getValue()` when you're already getting `Something` value. We just need to be consistent, if we have a `getProfileId` which return a ValueObject, all our `getProfileId` methods must return the same type.

It could have been cool if we returned a IntValueObject, UuuidValueObject or IdValueObject but not a ProfileId which is not really a type.

Reply:
What's the difference between a `IntValueObject` and a `ProfileId` for you ?
Is it an object that can be used as an integer, like a class with `__toString()` can be used as a string ?

Reply:
A `profile_id`, `shop_id`, `customer_id` are `IntValueObject`, but not `ProfileIdValueObject`, this is why this is actually really overkill because we repeat the same thing for all our fields :/

Reply:
I see, you mean that CustomerId, ProfileId or EmployeeId are all the same

Reply:
Yes, the ValueObject behind is the same

Reply:
I see this kind of discussion on internet pretty often. Some say that `Identity` (or similar named generic ID) should be used everywhere, while others defend using separate identities for different entities (e.g. `ProfileId`, `CustomerId`). 

For me, second option seems more explicit and less error prone than having just `Identity` VO for all entities (which we don't have in PS yet). Of course this comes with some code duplication, but I think we could solve it by abstracting some of it, for example:

```php
class ProfileId extends AbstractId
{
  // nothing here unless we have specific behavior to ProfileId
}
```

Reply:
👍 at least we're able to check for ValueObject is kind of IntValueObject / UuidValueObject and can help developers for a better understanding

Reply:
So do we need any changes here? :thinking:

Reply:
Waiting for @matks if he wants a ProfileId return or a int :)

Reply:
I'd prefer to stay with `int` for 1.7.6 as we are already in 1.7.6
And for 1.7.7 we can see whether we can introduce this `AbstractId` or `IntValueObject` and apply this in all DataProviderInterfaces, to stay consistent

Reply:
Issue created for that https://github.com/PrestaShop/PrestaShop/issues/13600